packages/hint-compat-api/docs/architecture.md

```diff
@@ -0,0 +1,26 @@
+# Architecture
+
+This hint groups 4 different hints:
+
+* Hint for deprecated CSS features.
+* Hint for not broadly supported CSS features.
+* Hint for deprecated HTML features.
+* Hint for not broadly supported HTML features.
+
+It relies on the data from https://github.com/mdn/browser-compat-data to
+know the status of CSS and HTML features:
+
+* CSS: https://github.com/mdn/browser-compat-data/tree/master/css
+* HTML: https://github.com/mdn/browser-compat-data/tree/master/html
+
+## CLI Flow
+
+Every single hint takes the provided browserlist by the user (or the default
+one) and the hint context to filter the list of features to test. 
+
+Once the previous step is done, the parser exposes sequentially all the features
+included in the resource (CSS or HTML) and it proceeds testing the feature if
+needed and not included in the `ignore` built-in list. All the errors reported 
+are temporary stored and consumed once the `scan::end` event is obtained.
+
+[![Compat API hint's architecture](images/architecture.png)](images/architecture.png)
```

Comment:
Can you add a class inheritance diagram? This is probably the most OO hint we have at the moment with a few layers of abstraction and it will be useful for the future. Thanks!